source/core/slang-zip-file-system.h

```diff
@@ -0,0 +1,35 @@
+#ifndef SLANG_ZIP_FILE_SYSTEM_H
+#define SLANG_ZIP_FILE_SYSTEM_H
+
+#include "slang-basic.h"
+
+#include "../../slang-com-ptr.h"
+
+namespace Slang
+{
+
+class CompressedFileSystem : public RefObject, public ISlangMutableFileSystem
+{
+public:
+
+    enum class CompressionType
+    {
+        BestSpeed,
+        BestCompression,
+    };
+
+        /// Get as an archive (that can be saved to disk)
+    virtual ArrayView<uint8_t> getArchive() = 0;
+        /// Set the compression - used for any subsequent items added
+    virtual void setCompressionType(CompressionType type) = 0;
+
+        /// Create a zip with the contents of data/size (the contents of a zip file)
+    static SlangResult createZip(const void* data, size_t size, RefPtr<CompressedFileSystem>& out);
```

Comment:
The naming here might confuse people. The `createZip` call isn't creating a zip archive or anything like that, but rather is creating a zip *file system* from the data in an archive. I guess at a call site it would be more clear, written `CompressedFileSystem::createZip`, but it would obviously be even more clear if it was written `ZipFileSystem::create()`. I assume we can't use the latter because `ZipFileSystem` is declared in a `.cpp` file to hide its implementation details.

(C++ is a bad language for the way that its `.h`/`.cpp` separation actually forces you to compromise on API design if you care about information hiding)